8_Largest_product.py

```diff
@@ -0,0 +1,32 @@
+def largest_products_series(limit, series):
+    # converts large number to a series of digits
+    series_digits = [int(d) for d in str(series)]
+    max_product = 1
+    for index, digit in enumerate(series_digits):
+
+        cut_series = series_digits[index:index+limit]
```

Comment:
Don't you run off the edge of the array here? If index is `len(series) - 1` you don't get an exception here?

Reply:
hmm yea i dont know why it ran

Reply:
I think python slice just handles it by itself

Reply:
Yeah looks like it:

```python
>>> a = [0, 1, 2, 3, 4]
>>> a[3:5]
[3, 4]
>>> a[4:10]
[4]
>>> a[19]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```

it's not good to depend on implicit behavior like this. Generally speaking you should be more rigorous about understanding and controlling your iteration range. What if Python4 changes this behavior? What if you need to write some loops in another language?

Reply:
Then it would've thrown an index out of bounds exception and I would have fixed it?